pkg/filter/block/filter.go

```diff
@@ -40,6 +41,12 @@ type Filter struct {
 
 // Filter lets no event pass
 func (f *Filter) Filter(evt event.Event) []event.Event {
+	if f == nil {
```

Comment:
In what case is f nil? Only in the unit tests?

Reply:
honestly i don't know, we do this in all filters based on Trevor's reference implementation, i just copied this over to the few filters lacking this snippet to make them all the same - you are probably right that this is an artificial case only occurring in certain unit tests